ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/clientlibs/layouts/iteminjector/js/itemapi.js

```diff
@@ -0,0 +1,272 @@
+        { amount: 7, name: 'days' },
+        { amount: 4.34524, name: 'weeks' },
+        { amount: 12, name: 'months' },
+        { amount: Number.POSITIVE_INFINITY, name: 'years' }
+    ];
+
+    var formatTimeAgo = function (date, DATE_FORMATTER) {
+        var duration = (date - new Date()) / 1000
+        for (let i = 0; i <= DIVISIONS.length; i++) {
+            const division = DIVISIONS[i];
+            if (Math.abs(duration) < division.amount) {
+              return DATE_FORMATTER.format(Math.round(duration), division.name);
+            }
+            duration /= division.amount;
+        }
+    };
+
+    var  NS = "cmp",
+         TEMPLATE = 'item-template',
+         ERROR_CLASS = 'cmp-portallister__item--error',
+         ATTR_PREFIX = "data-" + NS + "-hook-" + TEMPLATE;
+
+    // handler for operations
+    var operationHandlers = {};
+
+    var selectors = {
+        self: "[" + ATTR_PREFIX + "=\"item\"]",
+        container: "[" + ATTR_PREFIX + "=\"container\"]",
+        title: "[" + ATTR_PREFIX + "=\"itemTitle\"]",
+        formLink:  "[" + ATTR_PREFIX + "=\"formLink\"]",
+        description:  "[" + ATTR_PREFIX + "=\"description\"]",
+        includes:  "[" + ATTR_PREFIX + "=\"includes\"]",
+        thumbnail: "[" + ATTR_PREFIX + "=\"thumbnail\"]",
+        operations: "[" + ATTR_PREFIX + "=\"operations\"]",
+        timeinfo: "[" + ATTR_PREFIX + "=\"timeinfo\"]"
+    };
+
+    var addTextWithTooltip = function (element, text, tooltip) {
+        if (text) {
+            element.appendChild(document.createTextNode(text));
+            element.setAttribute("title", tooltip || text);
+        }
+    },
+    addDateWithTooltip = function (element, dateInMillis) {
+        var dt = new Date(Number.parseInt(dateInMillis)), tooltip = dt.toLocaleString(), text = dt.toDateString();
+        if (window.Intl) {
+            // not supported in I.E, so don't make relative dates there
+            // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat
+            // automatically handles browser locale and timezone
+            DATE_FORMATTER = DATE_FORMATTER || new Intl.RelativeTimeFormat(undefined, {
+                                                 numeric: 'auto'
+                                               });
+            text = formatTimeAgo(dt, DATE_FORMATTER);
+        }
+        element.appendChild(document.createTextNode(text));
+        element.setAttribute("title", tooltip);;
+    },
+    linkClickHandler = function (event) {
+        var link = event.currentTarget.getAttribute('link');
+        window.open(link, "_blank");
+    },
+    operationClickHandler = function (event) {
+        var operation_dataStr = event.currentTarget.getAttribute('operation_data'),
+            operation_data = JSON.parse(operation_dataStr);
+        operationHandlers[operation_data.name].call(this, event, operation_data);
+    };
+
+    class Menu {
```

Comment:
This won't work on IE.

Reply:
@sakshi-arora1 any reason for keeping it as Class or should we move this to a js object ?

Reply:
Moved to object notation